.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: Workspace Checks & Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Clippy and fmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v1
+      - name: Check Formatting
+        run: cargo fmt -- --check
+      - name: Check Clippy
+        run: cargo clippy -- -Dclippy::all
+
+  test:
+    needs: check
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
```

Comment:
Can we add Windows here?
(we can skip it if it's tricky)

Reply:
it can easily be added to CI, but not so sure it is windows compatible as of now

Reply:
I don't think we have windows binaries for sandbox yet unfortunately. @chefsale I remember Willem was talking to you about those. Are there any new updates on this?